sync-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -252,6 +268,32 @@ protected static boolean validFieldName(String fieldName) {
         return result;
     }
 
+    /**
+     * Based on the proposed index and the list of existing indexes, this method checks
+     * whether another index can be created.  Currently the limit for TEXT indexes is 1.
+     * JSON indexes are unlimited.
+     *
+     * @param index the proposed index
+     * @param existingIndexes the list of already existing indexes
+     * @return whether the index limit has been reached
+     */
+    @SuppressWarnings("unchecked")
+    protected static boolean indexLimitReached(Index index, Map<String, Object> existingIndexes) {
+        if (index.indexType.equalsIgnoreCase(Index.TEXT_TYPE)) {
+            for (String name : existingIndexes.keySet()) {
+                Map<String, Object> existingIndex = (Map<String, Object>) existingIndexes.get(name);
+                String type = (String) existingIndex.get("type");
+                if (type.equalsIgnoreCase("text") && !name.equalsIgnoreCase(index.indexName)) {
```

Comment:
It'd be nice if we validated prior to hitting SQL (in case we work out an amazing backend someday) but I think as long as there's a test for this situation, we can catch it when it happens.